packages/venia-concept/src/components/Header/header.js

```diff
@@ -20,11 +22,34 @@ class Header extends Component {
             searchTrigger: PropTypes.string,
             secondaryActions: PropTypes.string,
             toolbar: PropTypes.string
-        })
+        }),
+        // isRequired won't work because initially itemsCount is undefined
+        // seems page re-renders before cart getDetails() returns value
+        itemsCount: PropTypes.number
     };
 
+    /**
+     * Add fill attribute to cart icon
+     * for filled cart
+     *
+     * @returns {Object}
+     */
+
+    get cartIcon() {
+        const itemsCount = this.props.itemsCount || false;
+        return !itemsCount || itemsCount === 0 ? (
+            <Icon name="shopping-cart" attrs={{ stroke: '#212121' }} />
+        ) : (
+            <Icon
+                name="shopping-cart"
+                attrs={{ stroke: '#212121', fill: '#212121' }}
+            />
```

Comment:
Lastly on this function, since the only thing changing is whether or not we add a `fill` attribute, we could extract that piece of logic out and simplify things a bit:

What do you think of something like:

```jsx
get cartIcon() {
    const svgAttributes = {
        stroke: 'rgb(var(--venia-text))',
    };

    if (this.props.itemsCount > 0) {
        svgAttributes.fill = 'rgb(var(--venia-text))';
    }

    return (
        <Icon
            name="shopping-cart"
            attrs={svgAttributes}
        />
    );
}
```

?

Reply:
yeah much DRY'er ;-)